kotlin/configure-subnetwork-trace/src/main/java/com/esri/arcgisruntime/sample/configuresubnetworktrace/MainActivity.kt

```diff
@@ -0,0 +1,340 @@
+package com.esri.arcgisruntime.sample.configuresubnetworktrace
+
+import android.content.pm.ActivityInfo
+import android.os.Bundle
+import android.text.InputType
+import android.text.method.ScrollingMovementMethod
+import android.util.Log
+import android.view.View
+import android.widget.AdapterView
+import android.widget.ArrayAdapter
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.data.CodedValue
+import com.esri.arcgisruntime.data.CodedValueDomain
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.utilitynetworks.UtilityAttributeComparisonOperator
+import com.esri.arcgisruntime.utilitynetworks.UtilityCategoryComparison
+import com.esri.arcgisruntime.utilitynetworks.UtilityElement
+import com.esri.arcgisruntime.utilitynetworks.UtilityElementTraceResult
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetwork
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetworkAttribute
+import com.esri.arcgisruntime.utilitynetworks.UtilityNetworkAttributeComparison
+import com.esri.arcgisruntime.utilitynetworks.UtilityTier
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceAndCondition
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceConditionalExpression
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceConfiguration
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceOrCondition
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceParameters
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraceType
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraversability
+import com.esri.arcgisruntime.utilitynetworks.UtilityTraversabilityScope
+import kotlinx.android.synthetic.main.activity_main.*
+
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG: String = MainActivity::class.java.simpleName
+
+  private val utilityNetwork by lazy {
+    UtilityNetwork(getString(R.string.naperville_electric_url))
+  }
+
+  private var initialExpression: UtilityTraceConditionalExpression? = null
+  private var sourceTier: UtilityTier? = null
+  private var sources: List<UtilityNetworkAttribute>? = null
+  private var operators: Array<UtilityAttributeComparisonOperator>? = null
+  private var startingLocation: UtilityElement? = null
+  private var values: List<CodedValue>? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+
+    exampleTextView.movementMethod = ScrollingMovementMethod()
+
+    // create a utility network and wait for it to finish to load
+    utilityNetwork.loadAsync()
+    utilityNetwork.addDoneLoadingListener {
+      if (utilityNetwork.loadStatus == LoadStatus.LOADED) {
+        // populate spinners for network attribute comparison
+        sources =
```

Comment:
as we discussed, some more comments through here to help readability would be great, along with possibly separating the UI from our API into a different method if that seems doable.

Reply:
I'm hoping the added comments and .also block help indicate that its UI stuff happening with the populating of the sources list